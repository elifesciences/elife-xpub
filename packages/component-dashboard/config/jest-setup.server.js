const { db } = require('@pubsweet/db-manager')
const replaySetup = require('../../../test/helpers/replay-setup')

replaySetup('success')

// The pipeline complains at the default of 5s - so double it!
// https://jestjs.io/docs/en/jest-object#jestsettimeouttimeout
jest.setTimeout(10000)

jest.mock('@pubsweet/component-send-email', () => ({
  _sendPromise: null,
  mails: [],
  send(mailData) {
    this._sendPromise = new Promise((resolve, reject) => {
      this.mails.push(mailData)
      resolve()
    })
    return this._sendPromise
  },
  getMails() {
    return this.mails
  },
  clearMails() {
    this.mails = []
  },
}))

afterAll(() => db.destroy())
