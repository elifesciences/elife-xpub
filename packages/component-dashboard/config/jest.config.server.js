module.exports = {
  displayName: 'package-server',
  rootDir: '../',
  setupTestFrameworkScriptFile: '<rootDir>/config/jest-setup.server.js',
  testMatch: ['<rootDir>/server/**/*.test.js'],
  transform: {}, // turn off babel for server code
  transformIgnorePatterns: ['/node_modules/(?!@?pubsweet)'],
  testEnvironment: 'node',
}
