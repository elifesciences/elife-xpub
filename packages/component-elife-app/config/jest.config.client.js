module.exports = {
  displayName: 'component-elife-app',
  rootDir: '../',
  setupFilesAfterEnv: ['<rootDir>/config/jest-setup.client.js'],
  testMatch: ['<rootDir>/client/**/*.test.js'],
  transformIgnorePatterns: ['/node_modules/(?!@?pubsweet|xpub)'],
  moduleNameMapper: {
    '\\.s?css$': 'identity-obj-proxy',
  },
}
