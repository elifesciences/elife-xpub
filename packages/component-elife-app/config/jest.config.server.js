module.exports = {
  displayName: 'component-elife-app',
  rootDir: '../',
  setupFilesAfterEnv: ['<rootDir>/config/jest-setup.server.js'],
  testMatch: ['<rootDir>/server/**/*.test.js'],
  transform: {}, // turn off babel for server code
  transformIgnorePatterns: ['/node_modules/(?!@?pubsweet)'],
  testEnvironment: 'node',
}
