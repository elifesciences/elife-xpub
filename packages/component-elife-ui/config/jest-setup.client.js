const { Adapter } = require('enzyme-adapter-react-16')
const { Enzyme } = require('enzyme')

Enzyme.configure({ adapter: new Adapter() })

window.scrollTo = jest.fn()
