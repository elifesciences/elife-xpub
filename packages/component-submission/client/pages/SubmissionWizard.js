import React, { useState } from 'react'
import { compose, branch, renderComponent } from 'recompose'
import { Switch, Redirect } from 'react-router-dom'
import { Box, Flex } from '@rebass/grid'
import styled from 'styled-components'
import { th } from '@pubsweet/ui-toolkit'
import { Formik, Form } from 'formik'
import * as yup from 'yup'
import {
  ErrorPage,
  TrackedRoute,
} from '@elifesciences/component-elife-app/client'
import { Loading } from '@elifesciences/component-elife-ui/client/atoms'
import { Button } from '@pubsweet/ui'

import AuthorStep from './AuthorStepPage'
import FilesStep from './FilesStepPage'
import DetailsStep from './DetailsStepPage'
import EditorStep from './EditorsStepPage'
import DisclosureStep from './DisclosureStepPage'

import SubmissionSave from '../components/SubmissionSave'
import WizardSubmit from '../components/WizardSubmit'
import ProgressBar from '../components/ProgressBar'
import wizardWithGQL from '../graphql/wizardWithGQL'
import {
  parseInputToFormData,
  parseFormToOutputData,
  flattenObject,
  getErrorStepsFromErrors,
  convertArrayToReadableList,
} from '../utils'
import { STEP_NAMES } from '../utils/constants'

import wizardSchema, {
  authorSchema,
  filesSchema,
  submissionSchema,
  editorsSchema,
} from '../utils/ValidationSchemas'

const BoxNoMinWidth = styled(Box)`
  min-width: 0;
`

const ErrorMessage = styled.div`
  color: ${th('colorError')};
`

export const SubmissionWizard = ({
  data, // Contains data from an apollo GQL query
  match, // Comes from react router, contains information about the path
  history,
  updateManuscript, // injected by reCompose
  submitManuscript, // injected by reCompose
}) => {
  const getCurrentStepFromPath = () =>
    STEP_NAMES.map(step => step.toLowerCase()).indexOf(
      history.location.pathname.split('/')[3].toLowerCase(),
    )

  const [currentStep, setCurrentStep] = useState(getCurrentStepFromPath())
  const [isUploading, setIsUploading] = useState(false)
  const [submissionAttempted, setsubmissionAttempted] = useState(false)

  const isLastStep = () => currentStep === STEP_NAMES.length - 1
  const initialValues = parseInputToFormData(data.manuscript)
  const stepValidation = [
    authorSchema,
    filesSchema,
    submissionSchema,
    editorsSchema,
    wizardSchema,
  ]

  const handleSave = formValues =>
    updateManuscript({
      variables: { data: parseFormToOutputData(formValues) },
    })

  const handleSubmit = formValues =>
    submitManuscript({
      variables: { data: parseFormToOutputData(formValues) },
    })

  return (
    <Formik
      initialValues={initialValues}
      onSubmit={handleSubmit}
      render={formikProps => (
        <Form data-testid="submission-wizard-form">
          <SubmissionSave
            disabled={formikProps.isSubmitting}
            handleSave={handleSave}
            values={{
              ...formikProps.values,
              lastStepVisited: history.location.pathname,
            }}
          />
          <Flex>
            <BoxNoMinWidth flex="1 1 100%" mx={[0, 0, 0, '16.666%']}>
              <Box my={5}>
                <ProgressBar currentStep={currentStep} />
              </Box>
              <Switch>
                <TrackedRoute
                  path={`${match.path}/author`}
                  render={() => <AuthorStep {...formikProps} />}
                />
                <TrackedRoute
                  path={`${match.path}/files`}
                  render={() => (
                    <FilesStep
                      {...formikProps}
                      isUploading={isUploading}
                      setIsUploading={setIsUploading}
                    />
                  )}
                />
                <TrackedRoute
                  path={`${match.path}/details`}
                  render={() => <DetailsStep {...formikProps} />}
                />
                <TrackedRoute
                  path={`${match.path}/editors`}
                  render={() => <EditorStep {...formikProps} />}
                />
                <TrackedRoute
                  path={`${match.path}/disclosure`}
                  render={() => (
                    <DisclosureStep
                      isSubmissionAttempted={submissionAttempted}
                      {...formikProps}
                    />
                  )}
                />
                <Redirect
                  from="/submit/:id"
                  to={`/submit/${match.params.id}/author`}
                />
                <ErrorPage error="404: page not found" />
              </Switch>
              {!!Object.keys(formikProps.errors).length &&
                submissionAttempted &&
                isLastStep() && (
                  <ErrorMessage data-test-id="test-error-message">
                    We&apos;re sorry but there appears to be one or more errors
                    in your submission that require attention before you can
                    submit. Please use the back button to review the{' '}
                    {convertArrayToReadableList(
                      getErrorStepsFromErrors(formikProps.errors),
                    )}{' '}
                    steps before trying again.
                  </ErrorMessage>
                )}
              <Flex mt={6}>
                <Box mr={3}>
                  <Button
                    data-test-id="back"
                    disabled={currentStep === 0}
                    onClick={() => {
                      setCurrentStep(currentStep - 1)
                      history.push(
                        `${match.url}/${STEP_NAMES[
                          currentStep - 1
                        ].toLowerCase()}`,
                      )
                    }}
                  >
                    Back
                  </Button>
                </Box>
                <Box>
                  {currentStep === STEP_NAMES.length - 1 ? (
                    <WizardSubmit
                      setSubmissionAttempted={setsubmissionAttempted}
                      setTouched={formikProps.setTouched}
                      submitForm={formikProps.submitForm}
                      validateForm={formikProps.validateForm}
                    />
                  ) : (
                    <Button
                      data-testid="next"
                      onClick={() => {
                        formikProps.validateForm().then(errors => {
                          if (!Object.keys(errors).length) {
                            setCurrentStep(currentStep + 1)
                            history.push(
                              `${match.url}/${STEP_NAMES[
                                currentStep + 1
                              ].toLowerCase()}`,
                            )
                          }

                          Object.keys(errors).forEach(errorField => {
                            if (typeof errors[errorField] === 'object') {
                              const flattenedSubfields = flattenObject(
                                errors[errorField],
                              )
                              Object.keys(flattenedSubfields).forEach(
                                subField => {
                                  formikProps.setFieldTouched(
                                    `${errorField}.${subField}`,
                                    true,
                                  )
                                },
                              )
                            } else {
                              formikProps.setFieldTouched(
                                errorField,
                                true,
                                false,
                              )
                            }
                          })
                        })
                      }}
                      primary
                    >
                      Next
                    </Button>
                  )}
                </Box>
              </Flex>
            </BoxNoMinWidth>
          </Flex>
        </Form>
      )}
      validationSchema={yup.object().shape(stepValidation[currentStep])}
    />
  )
}

export default compose(
  wizardWithGQL, // This injects updateManuscript and submitManuscript into the props
  branch(
    props => props.data && (props.data.loading && !props.data.manuscripts),
    renderComponent(Loading),
  ),
  branch(props => !props.data || props.data.error, renderComponent(ErrorPage)),
  branch(
    props => props.data.manuscript.clientStatus !== 'CONTINUE_SUBMISSION',
    () => () => <Redirect to="/" />,
  ),
)(SubmissionWizard)
