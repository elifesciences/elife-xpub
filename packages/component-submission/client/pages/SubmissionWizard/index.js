import React from 'react'
import { Switch, Redirect } from 'react-router-dom'
import { Subscription } from 'react-apollo'
import { ErrorPage } from '@elifesciences/component-elife-app/client'

import SubmissionOperations from '../../components/SubmissionOperations'
import AuthorPageContainer from '../../components/steps/Author'
import FilesPageContainer from '../../components/steps/Files'
import SubmissionPage from '../../components/steps/Submission'
import EditorsPageContainer from '../../components/steps/Editors'
import DisclosurePage from '../../components/steps/Disclosure'
import { schema as authorPageSchema } from '../../components/steps/Author/schema'
import { schema as filesPageSchema } from '../../components/steps/Files/schema'
import { schema as submissionPageSchema } from '../../components/steps/Submission/schema'
import { schema as editorsPageSchema } from '../../components/steps/Editors/schema'
import { schema as disclosurePageSchema } from '../../components/steps/Disclosure/schema'
import WizardStep from '../../components/WizardStep'
import { ON_UPLOAD_PROGRESS } from '../../components/operations'
import TrackedRoute from '../../../../../app/client/trackedRoute'

const SubmissionWizard = ({ match, history }) => (
  <SubmissionOperations manuscriptId={match.params.id}>
    {({ initialValues, updateManuscript, submitManuscript }) => (
      <Subscription
        subscription={ON_UPLOAD_PROGRESS}
        variables={{ id: match.params.id }}
      >
        {({ data: uploadData, loading: uploadLoading }) => (
          <Switch>
            <TrackedRoute
              exact
              path={`${match.path}/files`}
              render={() => (
                <WizardStep
                  component={FilesPageContainer}
                  handleAutoSave={updateManuscript}
                  handleButtonClick={updateManuscript}
                  history={history}
                  initialValues={initialValues}
                  nextUrl={`${match.url}/submission`}
                  previousUrl={`${match.url}/author`}
                  step={1}
                  title="Write your cover letter and upload your manuscript"
                  uploadData={uploadData}
                  uploadLoading={uploadLoading}
                  validationSchema={filesPageSchema}
                />
              )}
            />
            <TrackedRoute
              path={`${match.path}/submission`}
              render={() => (
                <WizardStep
                  component={SubmissionPage}
                  handleAutoSave={updateManuscript}
                  handleButtonClick={updateManuscript}
                  history={history}
                  initialValues={initialValues}
                  nextUrl={`${match.url}/editors`}
                  previousUrl={`${match.url}/files`}
                  step={2}
                  title="Help us get your work seen by the right people"
                  validationSchema={submissionPageSchema}
                />
              )}
            />
            <TrackedRoute
              path={`${match.path}/editors`}
              render={() => (
                <WizardStep
                  component={EditorsPageContainer}
                  handleAutoSave={updateManuscript}
                  handleButtonClick={updateManuscript}
                  history={history}
                  initialValues={initialValues}
                  nextUrl={`${match.url}/disclosure`}
                  previousUrl={`${match.url}/submission`}
                  step={3}
                  title="Who should review your work?"
                  validationSchema={editorsPageSchema}
                />
              )}
            />
            <TrackedRoute
              path={`${match.path}/disclosure`}
              render={() => (
                <WizardStep
                  component={DisclosurePage}
                  finalStep
                  handleAutoSave={updateManuscript}
                  handleButtonClick={submitManuscript}
                  history={history}
                  initialValues={initialValues}
                  nextUrl={`/thankyou/${match.params.id}`}
                  previousUrl={`${match.url}/editors`}
                  step={4}
                  title="Disclosure of data to editors"
                  validationSchema={disclosurePageSchema}
                />
              )}
            />
            <TrackedRoute
              path={`${match.path}/author`}
              render={() => (
                <WizardStep
                  component={AuthorPageContainer}
                  handleAutoSave={updateManuscript}
                  handleButtonClick={updateManuscript}
                  history={history}
                  initialValues={initialValues}
                  nextUrl={`${match.url}/files`}
                  step={0}
                  title="Your details"
                  validationSchema={authorPageSchema}
                />
              )}
            />
            <Redirect
              exact
              from="/submit/:id"
              to={`/submit/${match.params.id}/author`}
            />
            <ErrorPage error="404: page not found" />
          </Switch>
        )}
      </Subscription>
    )}
  </SubmissionOperations>
)

export default SubmissionWizard
