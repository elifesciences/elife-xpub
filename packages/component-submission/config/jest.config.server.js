module.exports = {
  displayName: 'component-submission-server',
  rootDir: '../',
  setupTestFrameworkScriptFile: '<rootDir>/config/jest-setup.server.js',
  testMatch: ['<rootDir>/server/**/*.test.js'],
  transform: {}, // turn off babel for server code
  transformIgnorePatterns: ['/node_modules/(?!@?pubsweet)'],
  testEnvironment: 'node',
}
