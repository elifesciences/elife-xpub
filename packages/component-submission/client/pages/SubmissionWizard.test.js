/* eslint-disable import/first */
import { render, cleanup, fireEvent, configure } from '@testing-library/react'
import theme from '@elifesciences/elife-theme'
import React from 'react'
import 'jest-dom/extend-expect'
import flushPromises from 'flush-promises'
import config from 'config'
import { ThemeProvider } from 'styled-components'
import { MemoryRouter } from 'react-router-dom'
import { MockedProvider } from 'react-apollo/test-utils'
import { SubmissionWizard } from './SubmissionWizard'

jest.mock('react-ga')
jest.mock('./AuthorStepPage', () => () => 'AuthorStepPage')
jest.mock('./FilesStepPage', () => () => 'FilesStepPage')
jest.mock('./DetailsStepPage', () => () => 'DetailsStepPage')
jest.mock('./EditorsStepPage', () => () => 'EditorsStepPage')
jest.mock('./DisclosureStepPage', () => () => 'DisclosureStepPage')
jest.mock('../utils/ValidationSchemas', () => ({
  authorSchema: {},
  filesSchema: {},
  submissionSchema: {},
  editorsSchema: {},
  default: {},
}))

// needs extracting to own file #2126
const setupProvider = (historyLocation = []) => ({ children }) => (
  <ThemeProvider theme={theme}>
    <MemoryRouter initialEntries={historyLocation}>
      <MockedProvider>{children}</MockedProvider>
    </MemoryRouter>
  </ThemeProvider>
)

const makeProps = (path, pushHistory = jest.fn(), props) => ({
  data: { manuscript: { meta: {} } },
  match: { path: '/submit/id', url: '', params: { id: 'id' } },
  history: { location: { pathname: path }, push: pushHistory },
  updateManuscript: jest.fn(),
  submitManuscript: jest.fn(),
  ...props,
})

const renderWithPath = (path, pushHistory, props = {}) =>
  render(<SubmissionWizard {...makeProps(path, pushHistory, props)} />, {
    wrapper: setupProvider([path]),
  })

describe('SubmissionWizard', async () => {
  beforeAll(() => configure({ testIdAttribute: 'data-test-id' }))
  afterEach(cleanup)

  it('should display next on all steps except last one', async () => {
    expect(
      renderWithPath('/submit/id/author').getByText(/Next/),
    ).toBeInTheDocument()
    cleanup()
    expect(
      renderWithPath('/submit/id/files').getByText(/Next/),
    ).toBeInTheDocument()
    cleanup()
    expect(
      renderWithPath('/submit/id/details').getByText(/Next/),
    ).toBeInTheDocument()
    cleanup()
    expect(
      renderWithPath('/submit/id/editors').getByText(/Next/),
    ).toBeInTheDocument()
    cleanup()
    expect(
      renderWithPath('/submit/id/disclosure').getByText(/Submit/),
    ).toBeInTheDocument()
  })

  it('displays the correct step for each given path', () => {
    expect(
      renderWithPath('/submit/id/author').getAllByText('AuthorStepPage'),
    ).toBeTruthy()
    expect(
      renderWithPath('/submit/id/files').getAllByText('FilesStepPage'),
    ).toBeTruthy()
    expect(
      renderWithPath('/submit/id/details').getAllByText('DetailsStepPage'),
    ).toBeTruthy()
    expect(
      renderWithPath('/submit/id/editors').getAllByText('EditorsStepPage'),
    ).toBeTruthy()
    expect(
      renderWithPath('/submit/id/disclosure').getAllByText(
        'DisclosureStepPage',
      ),
    ).toBeTruthy()
  })

  describe('submit', () => {
    describe('survey feature flag', () => {
      const path = '/submit/id/disclosure'
      const submitManuscript = jest.fn()
      const pushHistory = jest.fn()
      const originalFeatures = { ...config.features }

      afterEach(() => {
        config.features = originalFeatures
        submitManuscript.mockReset()
        pushHistory.mockReset()
      })

      it('should redirect to thank you if survey flag is off', async () => {
        config.features.demographicSurvey = false
        const { getByTestId, getByText } = renderWithPath(path, pushHistory, {
          submitManuscript,
        })

        fireEvent.click(getByTestId('submit'))
        await flushPromises()

        fireEvent.click(getByText('Confirm'))
        await flushPromises()

        expect(submitManuscript).toHaveBeenCalledTimes(1)
        expect(pushHistory).toHaveBeenCalledTimes(1)
        expect(pushHistory).toHaveBeenCalledWith('/thankyou/id')
      })

      it('should redirect to survey page if survey flag is on', async () => {
        config.features.demographicSurvey = true
        const { getByTestId, getByText } = renderWithPath(path, pushHistory, {
          submitManuscript,
        })

        fireEvent.click(getByTestId('submit'))
        await flushPromises()

        fireEvent.click(getByText('Confirm'))
        await flushPromises()

        expect(submitManuscript).toHaveBeenCalledTimes(1)
        expect(pushHistory).toHaveBeenCalledTimes(1)
        expect(pushHistory).toHaveBeenCalledWith('/survey/id')
      })
    })
  })

  describe('step navigation', () => {
    // eslint-disable-next-line no-console
    const consoleError = console.error
    const pushHistory = jest.fn()
    const scrollSpy = jest.spyOn(window, 'scrollTo')

    const testStepNavigation = async (currentPath, nextStep, buttonId) => {
      const opts = renderWithPath(currentPath, pushHistory)
      fireEvent.click(opts.getByTestId(buttonId))
      await flushPromises()
      expect(pushHistory).toHaveBeenCalledTimes(1)
      expect(pushHistory.mock.calls[0]).toEqual([nextStep])
      cleanup()
      pushHistory.mockReset()
    }

    const SUPPRESSED_PREFIXES = [
      'Warning: Do not await the result of calling ReactTestUtils.act(...)',
      'Warning: An update to %s inside a test was not wrapped in act(...)',
    ]

    beforeAll(() => {
      // disable formik warnings
      // see https://stackoverflow.com/questions/55181009/jest-react-testing-library-warning-update-was-not-wrapped-in-act
      // https://github.com/facebook/react/pull/14853 removes the need for this once we upgrade to react-dom 16.9
      // eslint-disable-next-line no-console
      console.error = (...args) => {
        if (!SUPPRESSED_PREFIXES.some(sp => args[0].startsWith(sp))) {
          consoleError(...args)
        }
      }
    })

    it('should change the location on next', async () => {
      await testStepNavigation('/submit/id/author', '/files', 'next')
      await testStepNavigation('/submit/id/files', '/details', 'next')
      await testStepNavigation('/submit/id/details', '/editors', 'next')
      await testStepNavigation('/submit/id/editors', '/disclosure', 'next')
    })

    it('should change the location on back except on first step', async () => {
      const opts = renderWithPath('/submit/id/author')
      fireEvent.click(opts.getByTestId('back'))
      await flushPromises()
      expect(pushHistory).toHaveBeenCalledTimes(0)
      cleanup()
      pushHistory.mockReset()

      await testStepNavigation('/submit/id/files', '/author', 'back')
      await testStepNavigation('/submit/id/details', '/files', 'back')
      await testStepNavigation('/submit/id/editors', '/details', 'back')
      await testStepNavigation('/submit/id/disclosure', '/editors', 'back')
    })

    it('scrolls to top on next', async () => {
      const { getByTestId } = renderWithPath('/submit/id/author', pushHistory)

      expect(scrollSpy).toBeCalledTimes(0)

      fireEvent.click(getByTestId('next'))
      await flushPromises()

      expect(scrollSpy).toBeCalledTimes(1)
      expect(scrollSpy).toBeCalledWith(0, 0)
    })

    it('scrolls to top on back', async () => {
      const { getByTestId } = renderWithPath('/submit/id/files', pushHistory)

      expect(scrollSpy).toBeCalledTimes(0)

      fireEvent.click(getByTestId('back'))
      await flushPromises()

      expect(scrollSpy).toBeCalledTimes(1)
      expect(scrollSpy).toBeCalledWith(0, 0)
    })

    afterEach(() => {
      scrollSpy.mockClear()
    })

    afterAll(() => {
      // eslint-disable-next-line no-console
      console.error = consoleError
    })
  })

  describe('Step titles', () => {
    it('displays correct title on Author step', () => {
      expect(
        renderWithPath('/submit/id/author').getByText('Your details'),
      ).toBeInTheDocument()
    })
    it('displays correct title on Files step', () => {
      expect(
        renderWithPath('/submit/id/files').getByText('Your cover letter'),
      ).toBeInTheDocument()
    })
    it('displays correct title on Details step', () => {
      expect(
        renderWithPath('/submit/id/details').getByText(
          'Help us get your work seen by the right people',
        ),
      ).toBeInTheDocument()
    })
    it('displays correct title on Editors step', () => {
      expect(
        renderWithPath('/submit/id/editors').getByText(
          'Who should review your work?',
        ),
      ).toBeInTheDocument()
    })
    it('displays correct title on Disclosure step', () => {
      expect(
        renderWithPath('/submit/id/disclosure').getByText(
          'Disclosure of data to editors',
        ),
      ).toBeInTheDocument()
    })
  })
})
