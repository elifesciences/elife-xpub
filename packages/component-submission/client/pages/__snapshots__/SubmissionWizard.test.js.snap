// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionWizard renders the form 1`] = `
<div>
  <form
    data-testid="submission-wizard-form"
  >
    <div
      class="sc-bwzfXH sc-htpNat jNWwaj"
    >
      <div
        class="sc-bwzfXH sc-ugnQR loAYsM"
      >
        <div
          class="sc-bwzfXH iaRAcB"
        >
          <div
            class="sc-bwzfXH beuRlD"
            width="1,1,500"
          >
            <div
              class="sc-bwCtUz kQHPG"
            >
              <div
                class="sc-eXEjpC kefVwT"
              >
                <span
                  class="sc-iQKALj guhEvj"
                >
                  Author
                </span>
              </div>
              <div
                class="sc-kIPQKe bGehDG"
              />
              <div
                class="sc-eXEjpC kefVwT"
              >
                <span
                  class="sc-iQKALj guhEvj"
                >
                  Files
                </span>
              </div>
              <div
                class="sc-kIPQKe bGehDG"
              />
              <div
                class="sc-eXEjpC kefVwT"
              >
                <span
                  class="sc-iQKALj guhEvj"
                >
                  Details
                </span>
              </div>
              <div
                class="sc-kIPQKe bGehDG"
              />
              <div
                class="sc-eXEjpC kefVwT"
              >
                <span
                  class="sc-iQKALj guhEvj"
                >
                  Editors
                </span>
              </div>
              <div
                class="sc-kIPQKe bGehDG"
              />
              <div
                class="sc-eXEjpC kefVwT"
              >
                <span
                  class="sc-iQKALj guhEvj"
                >
                  Disclosure
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-eerKOB hdTJqZ"
        >
          <img
            alt="Test tube experiment"
            class="sc-emmjRN jkOBoT"
            src="/assets/error.svg"
          />
          <h1
            class="sc-jhAzac dcVnOl"
          >
            Oops!
          </h1>
          <h2
            class="sc-fBuWsC bTeJFU"
          >
            Something is wrong
          </h2>
          <pre>
            404: page not found
          </pre>
          <a
            href="/"
          >
            Back to homepage
          </a>
        </div>
        <div
          class="sc-bwzfXH sc-htpNat xGKBl"
        >
          <div
            class="sc-bwzfXH czAofS"
          >
            <button
              class="sc-dxgOiQ cowufr"
              data-test-id="back"
              type="button"
            >
              Back
            </button>
          </div>
          <div
            class="sc-bwzfXH ihgQcP"
          >
            <button
              class="sc-dxgOiQ gCHwDx"
              data-test-id="next"
              type="button"
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
